apps/site/src/app/(main)/login/Login.tsx

```diff
@@ -2,16 +2,27 @@ import { redirect } from "next/navigation";
 import getUserIdentity from "@/lib/utils/getUserIdentity";
 import LoginForm from "./components/LoginForm";
 
-async function Login() {
+async function Login({
+	searchParams,
+}: {
+	searchParams?: {
+		return_to?: string;
+	};
+}) {
 	const identity = await getUserIdentity();
-	if (identity.uid !== null) redirect("/portal");
+	if (identity.uid !== null) {
+		if (searchParams?.return_to === undefined) {
+			redirect("/portal");
+		}
+		redirect(`/portal?return_to=${searchParams.return_to}`);
+	}
```

Comment:
Question: same note as with guest login, if a user is already logged in, I don't think it's possible to navigate to `/login`, especially with a `return_to` value provided by us.

Reply:
Based on the other note, if the user spams the browser's navigates backwards (even past the SSO process), it is technically possible to end up here, so same logic:
```tsx
redirect(searchParams?.return_to ?? "/portal")
```